lib/Resque.php

```diff
@@ -11,196 +11,364 @@
+    {
+        self::$redisServer = $server;
+        self::$redisDatabase = $database;
+        self::$redis = null;
+        self::$namespace = $namespace;
+        self::$password = $password;
+    }
+
+    /**
+     * Return an instance of the Resque_Redis class instantiated for Resque.
+     *
+     * @return Resque_Redis Instance of Resque_Redis.
+     */
+    public static function redis()
+    {
+        // Detect when the PID of the current process has changed (from a fork, etc)
+        // and force a reconnect to redis.
+        $pid = getmypid();
+        if (self::$pid !== $pid) {
+            self::$redis = null;
+            self::$pid = $pid;
+        }
+
+        if (!is_null(self::$redis)) {
+            return self::$redis;
+        }
+
+        $server = self::$redisServer;
+        if (empty($server)) {
+            $server = 'localhost:6379';
+        }
+
+        if (is_array($server)) {
+            require_once dirname(__FILE__) . '/Resque/RedisCluster.php';
+            self::$redis = new Resque_RedisCluster($server);
+        } else {
+            if (strpos($server, 'unix:') === false) {
+                list($host, $port) = explode(':', $server);
+            } else {
+                $host = $server;
+                $port = null;
+            }
+            require_once dirname(__FILE__) . '/Resque/Redis.php';
+            $redisInstance = new Resque_Redis($host, $port, self::$password);
+            $redisInstance->prefix(self::$namespace);
+            self::$redis = $redisInstance;
+        }
+
+        if (!empty(self::$redisDatabase)) {
+            self::$redis->select(self::$redisDatabase);
+        }
+
+        return self::$redis;
+    }
+
+    /**
+     * Push a job to the end of a specific queue. If the queue does not
+     * exist, then create it as well.
+     *
+     * @param string $queue The name of the queue to add the job to.
+     * @param array $item Job description as an array to be JSON encoded.
+     */
+    public static function push($queue, $item)
+    {
+        self::redis()->sadd('queues', $queue);
+        self::redis()->rpush('queue:' . $queue, json_encode($item));
+    }
+
+    /**
+     * Pop an item off the end of the specified queue, decode it and
+     * return it.
+     *
+     * @param string $queue The name of the queue to fetch an item from.
+     * @return array Decoded item from the queue.
+     */
+    public static function pop($queue)
+    {
+        $item = self::redis()->lpop('queue:' . $queue);
+        if (!$item) {
+            return;
+        }
+
+        return json_decode($item, true);
+    }
+
+    /**
+     * Remove items of the specified queue
+     *
+     * @param string $queue The name of the queue to fetch an item from.
+     * @param array $items
+     * @return integer number of deleted items
+     */
+    public static function dequeue($queue, $items = [])
```

Comment:
I guess that `dequeue($queue)` (without $items) is equivalent to `clear(queue)` ?